<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Your Name" />

<meta name="date" content="2025-08-11" />

<title>SpatialGAMmod User Manual and Workflow</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">SpatialGAMmod User Manual and Workflow</h1>
<h4 class="author">Your Name</h4>
<h4 class="date">2025-08-11</h4>


<div id="TOC">
<ul>
<li><a href="#spatialgammod-user-manual-and-workflow" id="toc-spatialgammod-user-manual-and-workflow"><span class="toc-section-number">1</span> SpatialGAMmod: User Manual and
Workflow</a>
<ul>
<li><a href="#prerequisites" id="toc-prerequisites"><span class="toc-section-number">1.1</span> Prerequisites</a></li>
<li><a href="#assign-cells-to-spatial-contour-regions" id="toc-assign-cells-to-spatial-contour-regions"><span class="toc-section-number">1.2</span> Assign Cells to Spatial Contour
Regions</a></li>
<li><a href="#calculate-cell-type-proportions-per-region" id="toc-calculate-cell-type-proportions-per-region"><span class="toc-section-number">1.3</span> Calculate Cell Type Proportions
per Region</a></li>
<li><a href="#visualize-cell-type-proportions-as-heatmaps" id="toc-visualize-cell-type-proportions-as-heatmaps"><span class="toc-section-number">1.4</span> Visualize Cell Type Proportions as
Heatmaps</a></li>
<li><a href="#fit-spatial-gam-gradients-for-genes-or-signatures" id="toc-fit-spatial-gam-gradients-for-genes-or-signatures"><span class="toc-section-number">1.5</span> Fit Spatial GAM Gradients for
Genes or Signatures</a></li>
<li><a href="#predict-gam-expression-on-a-spatial-grid" id="toc-predict-gam-expression-on-a-spatial-grid"><span class="toc-section-number">1.6</span> Predict GAM Expression on a
Spatial Grid</a></li>
<li><a href="#compare-tumour-vs-non-tumour-regions" id="toc-compare-tumour-vs-non-tumour-regions"><span class="toc-section-number">1.7</span> Compare Tumour vs Non-Tumour
Regions</a></li>
<li><a href="#additional-analyses" id="toc-additional-analyses"><span class="toc-section-number">1.8</span> Additional Analyses</a>
<ul>
<li><a href="#spatial-autocorrelation-morans-i" id="toc-spatial-autocorrelation-morans-i"><span class="toc-section-number">1.8.1</span> Spatial Autocorrelation (Moran’s
I)</a></li>
<li><a href="#spatial-correlation-decay-with-distance" id="toc-spatial-correlation-decay-with-distance"><span class="toc-section-number">1.8.2</span> Spatial Correlation Decay with
Distance</a></li>
<li><a href="#run-pseudotime-within-regions" id="toc-run-pseudotime-within-regions"><span class="toc-section-number">1.8.3</span> Run Pseudotime Within
Regions</a></li>
</ul></li>
</ul></li>
</ul>
</div>

<div id="spatialgammod-user-manual-and-workflow" class="section level1" number="1">
<h1><span class="header-section-number">1</span> SpatialGAMmod: User
Manual and Workflow</h1>
<p>This document provides a step-by-step workflow to analyze spatial
transcriptomics data using the <strong>SpatialGAMmod</strong> package.
It guides you through key functions for assigning spatial regions,
modeling spatial gene expression gradients with GAMs, computing cell
type proportions, comparing tumor vs non-tumor regions, and visualizing
results.</p>
<hr />
<div id="prerequisites" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Prerequisites</h2>
<ul>
<li><p>A Seurat object with spatial transcriptomics data, including:</p>
<ul>
<li>Expression data (e.g., assay <code>&quot;RNA&quot;</code>, slot
<code>&quot;data&quot;</code>)</li>
<li>Spatial coordinates in metadata columns (default:
<code>&quot;imagecol&quot;</code> and <code>&quot;imagerow&quot;</code>)</li>
<li>Cell type annotations in metadata (e.g.,
<code>&quot;CellType&quot;</code>)</li>
</ul></li>
<li><p>The SpatialGAMmod package installed (see README for
installation).</p></li>
<li><p>Required R packages: <code>Seurat</code>, <code>mgcv</code>,
<code>ggplot2</code>, <code>viridis</code>, <code>dplyr</code>,
<code>tidyr</code>, <code>pheatmap</code>.</p></li>
</ul>
</div>
<div id="assign-cells-to-spatial-contour-regions" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Assign Cells to
Spatial Contour Regions</h2>
<p>Use spatial coordinates or clustering to assign cells into spatial
regions (contours) that represent biologically meaningful zones such as
tumor core, invasive front, stroma, etc.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(SpatialGAMmod)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co"># Assuming seurat_obj is pre-loaded</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">assign_contour_regions</span>(</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  seurat_obj,</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  <span class="at">coord_cols =</span> <span class="fu">c</span>(<span class="st">&quot;imagecol&quot;</span>, <span class="st">&quot;imagerow&quot;</span>),</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  <span class="at">num_contours =</span> <span class="dv">5</span>,</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;kmeans&quot;</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>)</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="co"># This adds a metadata column (default name &quot;assigned_region&quot;) to seurat_obj</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">head</span>(seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>assigned_region)</span></code></pre></div>
</div>
<div id="calculate-cell-type-proportions-per-region" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Calculate Cell Type
Proportions per Region</h2>
<p>Compute the proportions of each cell type within each spatial
region.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>prop_df <span class="ot">&lt;-</span> <span class="fu">calculate_celltype_proportions</span>(</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  seurat_obj,</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="at">region_col =</span> <span class="st">&quot;assigned_region&quot;</span>,</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="at">celltype_col =</span> <span class="st">&quot;CellType&quot;</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">head</span>(prop_df)</span></code></pre></div>
</div>
<div id="visualize-cell-type-proportions-as-heatmaps" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Visualize Cell Type
Proportions as Heatmaps</h2>
<p>Visualize these proportions across regions using heatmaps.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">proportion_heatmaps</span>(</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  prop_df,</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="at">region_col =</span> <span class="st">&quot;assigned_region&quot;</span>,</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">celltype_col =</span> <span class="st">&quot;CellType&quot;</span>,</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">proportion_col =</span> <span class="st">&quot;proportion&quot;</span>,</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="fit-spatial-gam-gradients-for-genes-or-signatures" class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> Fit Spatial GAM
Gradients for Genes or Signatures</h2>
<p>Model spatial gradients of gene expression or gene signatures using
Generalized Additive Models (GAMs). This step smooths expression over
spatial coordinates and optionally fits separate models per group (e.g.,
tumor cores).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>genes_to_model <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ACTA2&quot;</span>, <span class="st">&quot;COL1A1&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>signature_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">my_signature =</span> <span class="fu">c</span>(<span class="st">&quot;ACTA2&quot;</span>, <span class="st">&quot;COL1A1&quot;</span>, <span class="st">&quot;FAP&quot;</span>)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>gam_res <span class="ot">&lt;-</span> <span class="fu">spatial_gam_gradient</span>(</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>  seurat_obj,</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>  <span class="at">genes =</span> genes_to_model,</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>  <span class="at">signature_list =</span> signature_list,</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>  <span class="at">split_by =</span> <span class="st">&quot;tumor_core_location&quot;</span>,  <span class="co"># optional grouping</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  <span class="at">spatial_coords =</span> <span class="fu">c</span>(<span class="st">&quot;imagecol&quot;</span>, <span class="st">&quot;imagerow&quot;</span>),</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  <span class="at">plot_results =</span> <span class="cn">TRUE</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>)</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co"># View GAM fit summaries</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="fu">print</span>(gam_res<span class="sc">$</span>metadata)</span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co"># Plot smoothed spatial expression gradients (example for first plot)</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="fu">print</span>(gam_res<span class="sc">$</span>plots[[<span class="dv">1</span>]])</span></code></pre></div>
</div>
<div id="predict-gam-expression-on-a-spatial-grid" class="section level2" number="1.6">
<h2><span class="header-section-number">1.6</span> Predict GAM
Expression on a Spatial Grid</h2>
<p>Generate smooth spatial predictions for visualization or further
analysis:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># Choose a GAM model from the list returned by spatial_gam_gradient()</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>gam_model <span class="ot">&lt;-</span> gam_res<span class="sc">$</span>gam_fits[[<span class="dv">1</span>]]</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co"># Predict expression on a spatial grid</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>grid_pred <span class="ot">&lt;-</span> <span class="fu">predict_gam_to_grid</span>(</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  gam_model,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  seurat_obj<span class="sc">@</span>meta.data,</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="at">coord_cols =</span> <span class="fu">c</span>(<span class="st">&quot;imagecol&quot;</span>, <span class="st">&quot;imagerow&quot;</span>),</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="at">grid_length =</span> <span class="dv">100</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>)</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co"># Visualize grid predictions</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="fu">ggplot</span>(grid_pred, <span class="fu">aes</span>(<span class="at">x =</span> imagecol, <span class="at">y =</span> imagerow, <span class="at">fill =</span> predicted)) <span class="sc">+</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>  <span class="fu">geom_raster</span>() <span class="sc">+</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Predicted Spatial Expression&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
</div>
<div id="compare-tumour-vs-non-tumour-regions" class="section level2" number="1.7">
<h2><span class="header-section-number">1.7</span> Compare Tumour vs
Non-Tumour Regions</h2>
<p>Perform differential expression analysis between tumor and non-tumor
regions defined by contour assignments.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>diff_res <span class="ot">&lt;-</span> <span class="fu">tumour_vs_non_tumour_contour</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  seurat_obj,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">contour_col =</span> <span class="st">&quot;assigned_region&quot;</span>,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="at">tumour_regions =</span> <span class="fu">c</span>(<span class="st">&quot;tumour_core&quot;</span>),</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  <span class="at">non_tumour_regions =</span> <span class="fu">c</span>(<span class="st">&quot;stroma&quot;</span>, <span class="st">&quot;immune&quot;</span>),</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  <span class="at">genes =</span> genes_to_model,</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>  <span class="at">plot_results =</span> <span class="cn">TRUE</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>)</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co"># Inspect results</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="fu">head</span>(diff_res<span class="sc">$</span>results)</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co"># Plot volcano plot</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="fu">print</span>(diff_res<span class="sc">$</span>plot)</span></code></pre></div>
</div>
<div id="additional-analyses" class="section level2" number="1.8">
<h2><span class="header-section-number">1.8</span> Additional
Analyses</h2>
<div id="spatial-autocorrelation-morans-i" class="section level3" number="1.8.1">
<h3><span class="header-section-number">1.8.1</span> Spatial
Autocorrelation (Moran’s I)</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>moran_i <span class="ot">&lt;-</span> <span class="fu">spatial_autocorrelation</span>(</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  seurat_obj,</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="at">gene =</span> <span class="st">&quot;ACTA2&quot;</span>,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="at">coord_cols =</span> <span class="fu">c</span>(<span class="st">&quot;imagecol&quot;</span>, <span class="st">&quot;imagerow&quot;</span>),</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="at">neighbors_k =</span> <span class="dv">6</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>)</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="fu">print</span>(moran_i)</span></code></pre></div>
</div>
<div id="spatial-correlation-decay-with-distance" class="section level3" number="1.8.2">
<h3><span class="header-section-number">1.8.2</span> Spatial Correlation
Decay with Distance</h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>dist_corr <span class="ot">&lt;-</span> <span class="fu">spatial_correlation_with_distance</span>(</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  seurat_obj,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">gene =</span> <span class="st">&quot;ACTA2&quot;</span>,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="at">coord_cols =</span> <span class="fu">c</span>(<span class="st">&quot;imagecol&quot;</span>, <span class="st">&quot;imagerow&quot;</span>),</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>  <span class="at">n_bins =</span> <span class="dv">20</span>,</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>  <span class="at">sample_size =</span> <span class="dv">10000</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>)</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="fu">plot</span>(dist_corr<span class="sc">$</span>mean_distance, dist_corr<span class="sc">$</span>mean_correlation, <span class="at">type =</span> <span class="st">&quot;b&quot;</span>,</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Distance&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Expression similarity&quot;</span>,</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;Spatial correlation decay with distance&quot;</span>)</span></code></pre></div>
</div>
<div id="run-pseudotime-within-regions" class="section level3" number="1.8.3">
<h3><span class="header-section-number">1.8.3</span> Run Pseudotime
Within Regions</h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">run_pseudotime_within_region</span>(</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  seurat_obj,</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="at">region_col =</span> <span class="st">&quot;assigned_region&quot;</span>,</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>,</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  <span class="at">cluster_col =</span> <span class="st">&quot;CellType&quot;</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>)</span></code></pre></div>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
